action.yml

```diff
@@ -1,60 +1,188 @@
 inputs:
   app_name:
-    description: 'Application name to test. E.g., eox-tenant.'
+    description: "Open edX plugin or application name to test. E.g., eox-tenant."
     required: true
   tutor_version:
-    description: 'The tutor version matrix to use.'
+    description: "The tutor version matrix to use."
     required: true
   shell_file_to_run:
-    description: 'The path of the shell file to run the integration tests.'
+    description: "The path of the shell file to run the integration tests."
+    required: true
+  openedx_extra_pip_requirements:
+    description: "Optional extra pip requirements to install in Open edX. E.g: 'package1==1.0 package2>=2.0'"
+    required: false
+    default: ""
+  fixtures_file:
+    description: "Optional path to the plugin's fixtures file to load."
+    required: false
+  openedx_imports_test_file_path:
+    description: "Path to the file that contains the test function for validating Open edX imports. This should be a Python file within your project."
+    required: false
+  openedx_imports_test_function_name:
+    description: "Name of the function in the specified file that executes the import tests for Open edX."
     required: false
 
 runs:
-  using: 'composite'
+  using: "composite"
   steps:
-  - name: Create a Virtual Environment
-    run: |
-      pip install virtualenv
-      virtualenv venv
-    shell: bash
-
-  - name: Install Tutor
-    run: |
-      source venv/bin/activate
-      if [ "$INPUT_TUTOR_VERSION" = "nightly" ]; then
-        git clone --branch=nightly --depth=1 https://github.com/overhangio/tutor.git
-        chmod 777 tutor -R
-        pip install -e "./tutor[full]"
-      else
-        pip install "tutor$INPUT_TUTOR_VERSION"
-      fi
-    shell: bash
-    env:
-      INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
-
-  - name: Configure Tutor and Launch
-    run: |
-      source venv/bin/activate
-      TUTOR_ROOT="$(pwd)" tutor --version
-      TUTOR_ROOT="$(pwd)" tutor config save
-      TUTOR_ROOT="$(pwd)" tutor mounts add lms,cms,lms-worker,cms-worker:$(pwd)/$INPUT_APP:/openedx/$INPUT_APP
-      chmod 777 . -R
-      if [ "$INPUT_TUTOR_VERSION" = "nightly" ]; then
-        TUTOR_ROOT="$(pwd)" tutor images build openedx
-      fi
-      TUTOR_ROOT="$(pwd)" tutor local launch -I
-    shell: bash
-    env:
-      INPUT_APP: ${{ inputs.app_name }}
-      INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
-
-  - name: Run integration tests in lms
-    if: ${{ inputs.shell_file_to_run }}
-    run: |
-      source venv/bin/activate
-      TUTOR_ROOT="$(pwd)" tutor local run lms bash $INPUT_SHELL_FILE
-    shell: bash
-    env:
-      INPUT_SHELL_FILE: /openedx/${{ inputs.app_name }}/${{ inputs.shell_file_to_run }}
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        path: ${{ inputs.app_name }}
+
+    - name: Adjust permissions to execute Tutor commands
+      run: |
+        chmod 777 . -R
```

Comment:
This question is totally out of scope, considering it was a design decision made and approved in a previous PR. Please feel free to reference any previous discussions in case I've forgotten or missed them.

We had [this workflow](https://github.com/eduNEXT/tutor-local-nightly-athena/blob/main/.github/workflows/build.yml) to build nightly images that didn't need permission adjustments. What's the difference between this GH action and the workflow?

Reply:
Please, see: https://github.com/eduNEXT/integration-test-in-tutor/pull/4#discussion_r1783558868